src/common/rand/rand.h

```diff
@@ -66,6 +68,21 @@ OQS_API void OQS_randombytes(uint8_t *random_array, size_t bytes_to_read);
  */
 OQS_API void OQS_randombytes_nist_kat_init_256bit(const uint8_t *entropy_input, const uint8_t *personalization_string);
 
+/**
+ * Initializes the HQC PRNG with a given seed.
+ * OQS_randombytes_hqc_kat_release should be called afterwards to clean up memory.
+ *
+ * @param[in] entropy_input The seed; must be exactly 48 bytes
+ * @param[in] personalization_string An optional personalization string;
+ * may be NULL; if not NULL, must be at least 48 bytes long
+ */
+OQS_API void OQS_randombytes_hqc_kat_init(const uint8_t *entropy_input, const uint8_t *personalization_string);
```

Comment:
Is this change to/extension of the "common" `liboqs` API really necessary? It feels wrong to add an algorithm-specific API call... Couldn't this be "hidden" within the tests code? Do we assume others will need this?

Reply:
We could define these functions within the `kat_kem.c` file and use the `OQS_randombytes_custom_algorithm` function to hook them in. I have no objection to doing that. In adding them to the API I was just following the same pattern as we use for the NIST KAT DRBG. Although that is presumably of much wider interest than the HQC version. The only use case that I can think of for someone else needing to have the HQC PRNG interface exposed is to run all 100 KATs, since our test currently only checks the first. And we'll (hopefully) be fixing that soon.

The HQC code is "public domain" so presumably the SPDX identifier for `kat_kem.c` (currently MIT) would need to be updated?

Reply:
> I have no objection to doing that. 

That would be much appreciated and keep the `liboqs` API clean.

> NIST KAT DRBG. Although that is presumably of much wider interest than the HQC version

It'd also think so.

> The HQC code is "public domain" so presumably the SPDX identifier for kat_kem.c (currently MIT) would need to be updated?

Hmm -- is a call to set up an RNG in the way HQC needs it truly HQC code / subject to its license?

Reply:
As discussed: The "MIT way" currently stated in "tests/kat_kem.c" is correct. Thanks for moving the code there. This resolves my concern.